guides/hack/03-async/01-intro.md

```diff
@@ -1,38 +1,32 @@
-
-In the example above, the call to `curl_exec` in `curl_A()` is blocking any other processing. Thus, even though `curl_B()` is an independent call from `curl_A()`, it has to sit around waiting for `curl_A()` to finish before beginning its execution.
-
-![No Async](/images/no-async.png)
-
 Hack provides a feature called **async** that provides your program the benefit of cooperative multi-tasking. It allows code that utilizes the async infrastructure to hide input/output (I/O) latency and data fetching. So, if you have code that has operations that involve some sort of waiting (e.g., network access, waiting for database queries), async minimizes the downtime your program has to be stalled because of it as the program will go do other things, most likely further I/O somewhere else.
 
-Async is **not multithreading**. You are still bound to a single execution thread. Async works best when you have a lot of I/O codepaths that don't have to sit there waiting for other I/O requests to end to begin doing their requests.
+Async is **not multithreading** - HHVM still executes all of your PHP/Hack code in one main request thread - but other operations (eg MySQL queries) can now execute without taking up time in that thread that your code could be using.
 
-The following images should hopefully clear up any confusion you may have on what async is. Let's assume you have three distinct tasks to execute (don't worry about what the tasks are):
+## A Page As A Dependency Tree
 
-## Synchronous Execution
+Imagine you have a page that contains two components; one stores data in MySQL, the other fetches from an API via CuRL - and both cache results in Memcached. The dependencies could be modeled like this:
 
-This is just like PHP and Hack (without async) is executed today. Serial execution.
+![Dependency Tree](/images/async/async-dependency.png)
 
-![Synchronous](/images/synchronous.png)
+Code structured like this gets the most benefit from async.
 
-## Parallel Execution
+## Synchronous/Blocking IO: Sequential Execution
 
-This is an optimum state. We have all tasks running at the same time, concurrently. But, PHP and Hack do not support more than one thread of execution.
+If (like most PHP code) you do not use asynchronous programming, each step will be executed one-after-the-other:
 
-![Parallel](/images/parallel.png)
+![Sequential Execution](/images/async/async-sequential.png)
 
 ## Asynchronous Execution
 
-This is what async does. Tasks are executed concurrently in the same execution thread, with respect to each other, interleaving instructions (e.g., I/O) for different tasks back and forth.
+This is what Hack's async functionality is. All PHP/Hack code executes in the main request thread, but I/O does not block it, and multiple I/O or other async tasks can execute concurrently. If your code is constructed as a dependency tree, this will lead to various parts of your code transparently interleaving with each other instead of blocking each other:
```

Comment:
This phrasing is awkward, especially after removing the "multithreaded" section (the removal of which is good :)). This implies that Hack gets async for free. Maybe something about properly-written async Hack code would look like this?